src/tasks.ts

```diff
@@ -173,27 +179,36 @@ export async function getMesonTasks(buildDir: string): Promise<vscode.Task[]> {
 export async function getTask(mode: string, name?: string) {
   const tasks = await vscode.tasks.fetchTasks({ type: "meson" });
   const filtered = tasks.filter(
-    t => t.definition.mode === mode && (!name || t.definition.target === name)
+    (t) => t.definition["mode"] === mode && !name || t.definition["target"] === name
   );
-  if (filtered.length === 0)
+  if (filtered.length === 0) {
     throw new Error(`Cannot find ${mode} target ${name}.`);
+  }
+
   return filtered[0];
 }
 
 export async function getTasks(mode: string) {
   const tasks = await vscode.tasks.fetchTasks({ type: "meson" });
   return tasks.filter(
-    t => t.definition.mode === mode
+    (t) => t.definition["mode"] === mode
   );
 }
 
 export async function runTask(task: vscode.Task) {
   try {
     await vscode.tasks.executeTask(task);
-  } catch (e) {
-    vscode.window.showErrorMessage(`Could not ${task.definition.mode} ${task.name}`);
+  } catch (e: any) {
```

Comment:
Any better type for this?

Reply:
Sadly not, but that's TS exceptions for you! Can't specify a type other than any (the default with strict `unknown`  due to `useUnknownInCatchVariables`).

So this just introspects `e` since it seemed pretty unspecified as to what exception type might be caught here (if any...). Plenty of other approaches for somewhat type-safe exception handling, but it's no worse than it was, hopefully, so didn't seem worth going crazy.